brian2/tests/test_codegen.py

```diff
@@ -389,19 +389,19 @@ def test_automatic_augmented_assignments():
         # examples that should be rewritten
         # Note that using our approach, we will never get -= or /= but always
         # the equivalent += or *= statements
-        ('x = x + 1', 'x += 1'),
-        ('x = 2 * x', 'x *= 2'),
-        ('x = x - 3', 'x += -3'),
-        ('x = x/2', 'x *= 0.5'),
-        ('x = y + (x + 1)', 'x += y + 1'),
-        ('x = x + x', 'x *= 2'),
+        ('x = x + 1.0', 'x += 1.0'),
+        ('x = 2.0 * x', 'x *= 2.0'),
+        ('x = x - 3.0', 'x += -3.0'),
+        ('x = x/2.0', 'x *= 0.5'),
+        ('x = y + (x + 1.0)', 'x += y + 1.0'),
+        ('x = x + x', 'x *= 2.0'),
         ('x = x + y + z', 'x += y + z'),
         ('x = x + y + z', 'x += y + z'),
         # examples that should not be rewritten
-        ('x = 1/x', 'x = 1/x'),
-        ('x = 1', 'x = 1'),
-        ('x = 2*(x + 1)', 'x = 2*(x + 1)'),
-        ('x = clip(x + y, 0, inf)', 'x = clip(x + y, 0, inf)'),
+        ('x = 1.0/x', 'x = 1.0/x'),
+        ('x = 1.0', 'x = 1.0'),
+        ('x = 2.0*(x + 1.0)', 'x = 2.0*(x + 1.0)'),
+        ('x = clip(x + y, 0.0, inf)', 'x = clip(x + y, 0.0, inf)'),
```

Comment:
Wait, I'm confused: why were these integers before and now floats?

Reply:
Yea, this is confusing... The explication is: we are using sympy behind the scenes in this test to compare the optimized expressions. This is to avoid that `x + 1` is considered as different from `1 + x`, for example.

Here we are testing the codegen optimization and during this process we do sometimes replace integers by floats as a side-effect (e.g. `x + 1` will turn into `x + 1.0` if `x` is a float). This is fine and nothing to worry about, but it made the test fail due to sympy making a difference between the two expressions now. When we are testing that `x = 1/x` does not change during optimization, we are not caring about whether the `1` continues to be an integer – instead of making the test comparison more complicated, I simply turned all values into floats...

Reply:
Ahhh I see. OK, in that case I'm all good! Happy to merge.